spec/lucky/action_spec.cr

```diff
@@ -145,6 +164,64 @@ describe Lucky::Action do
       action = PlainText::Index.new(build_context(path: "/?q=test"), params)
       action.params.get(:q).should eq "test"
     end
+
+    it "can get manually defined required params" do
+      action = RequiredParams::Index.new(build_context(path: "/?required_page=1"), params)
+      action.required_page.should eq 1
+    end
+
+    it "raises for missing required params" do
+      action = RequiredParams::Index.new(build_context(path: ""), params)
+      expect_raises(Lucky::Exceptions::MissingParam) { action.required_page }
+    end
+  end
+
+  describe "optional params" do
+    it "is initialized to nil" do
+      action = OptionalParams::Index.new(build_context(path: ""), params)
+      action.page.should eq nil
+    end
+
+    it "is fetched if present" do
+      action = OptionalParams::Index.new(build_context(path: "/?page=3"), params)
+      action.page.should eq 3
+    end
+
+    it "can be used within the action" do
+      response = OptionalParams::Index.new(build_context(path: "/?page=3"), params).call
+      response.body.to_s.should eq "optional param: 3 1 1337"
+    end
+
+    it "can specify a default value" do
+      action = OptionalParams::Index.new(build_context(path: ""), params)
+      action.with_default.should eq "default"
+    end
+
+    it "overrides the default if present" do
+      action = OptionalParams::Index.new(build_context(path: "/?with_int_never_nil=42"), params)
+      action.with_int_never_nil.should eq 42
+    end
+
+    it "is added as optional argument to the path" do
+      OptionalParams::Index.path(7).should eq "/optional_params?page=7"
+      OptionalParams::Index.path(7, "/other").should eq "/optional_params?page=7&with_default=%2Fother"
```

Comment:
Could you use the named arg version so it's clear what is going on `path(page: 7, with_default: "/other"`?